.github/workflows/test-activate-venv.yml

```
@@ -17,35 +17,97 @@ concurrency:
+    container: ${{ matrix.os.container[matrix.python.matrix] }}
     strategy:
       fail-fast: false
       matrix:
         os:
           - name: macOS
             matrix: macos
-            runs-on: macos-latest
+            runs-on:
+              arm: [macOS, ARM64]
+              intel: [macos-latest]
           - name: Ubuntu
             matrix: ubuntu
-            runs-on: ubuntu-latest
-          - name: Ubuntu Docker
-            matrix: docker
-            runs-on: ubuntu-latest
-            container: ubuntu
+            runs-on:
+              arm: [Linux, ARM64]
+              intel: [ubuntu-latest]
+          - name: Docker Debian
+            matrix: docker-debian
+            runs-on:
+              arm: [Linux, ARM64]
+              intel: [ubuntu-latest]
+            container:
+              'py3.7': docker://python:3.7-bullseye
+              'py3.8': docker://python:3.8-bullseye
+              'py3.9': docker://python:3.9-bullseye
+              'py3.10': docker://python:3.10-bullseye
+              'py3.11': docker://python:3.11-bullseye
           - name: Windows
             matrix: windows
-            runs-on: windows-latest
+            runs-on:
+              intel: [windows-latest]
         python:
+          - name: '3.7'
+            action: '3.7'
+            check: '3.7'
+            matrix: 'py3.7'
+          - name: '3.8'
+            action: '3.8'
+            check: '3.8'
+            matrix: 'py3.8'
           - name: '3.9'
             action: '3.9'
+            check: '3.9'
+            matrix: 'py3.9'
+          - name: '3.10'
+            action: '3.10'
+            check: '3.10'
+            matrix: 'py3.10'
+          - name: '3.11'
+            action: '3.11'
+            check: '3.11'
+            matrix: 'py3.11'
+        arch:
+          - name: ARM
+            matrix: arm
+          - name: Intel
+            matrix: intel
+        exclude:
+          # Only partial entries are required here by GitHub Actions so generally I
+          # only specify the `matrix:` entry.  The super linter complains so for now
+          # all entries are included to avoid that.  Reported at
+          # https://github.com/rhysd/actionlint/issues/249
+          - os:
+              name: Windows
+              matrix: windows
+              runs-on:
+                intel: [windows-latest]
+            arch:
+              name: ARM
+              matrix: arm
+          - os:
+              name: macOS
+              matrix: macos
+              runs-on:
+                arm: [macOS, ARM64]
+                intel: [macos-latest]
+            python:
+              name: '3.7'
+              action: '3.7'
+              check: '3.7'
+              matrix: 'py3.7'
+            arch:
+              name: ARM
+              matrix: arm
 
     steps:
     - uses: actions/checkout@v3
 
     - name: Set up ${{ matrix.python.name }}
-      uses: actions/setup-python@v3
+      uses: ./setup-python
```

```suggestion
      uses: ${{ env.GITHUB_ACTION_PATH }}/setup-python
```

Wondering if we should use this instead? I think this will work in external repos as well, and the `env` variant I think should be correct inside docker typically

---

I considered it briefly but figured this was a test so I didn't need to figure that out now.

---

Fair point, missed that this was in a test. If we swap to testing without checkouts later, we'll probably need something like this then